client/src/page/AbsenceSchedule.jsx

```diff
@@ -10,31 +10,11 @@ import {checkRole} from '../Helper/Auth';
 import {UserContext} from '../App'
 import { Stack } from '@mui/material';
 
-export default function AbsenceSchedule() {
-  //TODO OCT 26:
-  // Make it not ugly
-  // Styling and redoing period display
-  // Handle date picker differently (or remove)
-  // Investigate phantom absences?
-
 
-  // IVYSAUR-53 required changes:
-  //  Include WeekControl
-  //  Teacher table should update on change to current week
-  //  Should automatically load current week's absences from DB on component mount
-  //  Filterable (e.g. by teacher name):
-  //      Search bar to the left of teacher table
-  //      autocomplete suggestions as possible quality-of-life improvement, but depends on effort required
-  //  Each teacher needs an edit ('✏') button:
-  //      Edit button opens teacherAbsences component
-  //      Controls for setting absences should only be enabled for future periods
-  //      After saving changes to teacherAbsences table is updated to match
-  //  Separate OA and Teacher functionality:
-  //      For OA: All teachers can be edited
-  //      For teacher: View is restricted to their own absences
 
-  checkRole(useContext(UserContext).role,['ADMIN']);
-  const [selectedFile, setSelectedFile] = useState(null);
+export default function AbsenceSchedule() {
+  const user = useContext(UserContext);
+  checkRole(user.role,['ADMIN','TEACHER']);
```

Comment:
My bad. I should have removed this. I don't think we need to check roles because the new dashboard makes it so users can only use elements we let them see.

Reply:
I get what you're saying but I think it's fine to leave in. This way if a user without access is given a link to this page they will get redirected